lib/pbench/agent/tool_meister.py

```diff
@@ -1311,125 +1355,88 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
+
         try:
-            # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
-                )
+            cp = self._create_tar(directory, tar_file, retry=True)
+            if cp.returncode != 0:
+                self.logger.warning("Failed to create tarball, %s", cp.stdout)
+                # Tar ball creation failed even after suppressing all the warnings,
+                # we will now proceed to create an empty tar ball.
+                # TODO: it'd be better to be able to skip the PUT entirely if the
+                # tar fails and simply log a failure without TDS waiting forever.
+                if self._create_tar(Path("/dev/null"), tar_file) != 0:
+                    # Empty tarball creation failed, so we're going to skip the PUT
+                    # operation, and that's going to cause the TDS to hang.
+                    raise ToolMeisterError("Failed to create an empty tar")
         except Exception:
-            self.logger.exception("Failed to create tar ball '%s'", tar_file)
+            self.logger.exception("Failed to create tar ball, '%s'", tar_file)
             failures += 1
         else:
             try:
-                if cp.returncode != 0:
-                    self.logger.error(
-                        "Failed to create tar ball; return code: %d",
-                        cp.returncode,
-                    )
-                    failures += 1
-                else:
+                (_, tar_md5) = md5sum(tar_file)
+            except Exception:
+                self.logger.exception("Failed to read tar ball, '%s'", tar_file)
```

Comment:
Yes!  What Dave said:  the exception will supply the "what happened" and/or "why it failed"; what we need the log message to provide is the context of what we were trying to do at the time.  So, yeah, the value of `tar_file` is key, and it would be good to mention that we had called `md5sum()` (although that might be apparent from the traceback).